hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreFlusher.java

```diff
@@ -467,18 +469,28 @@ public boolean requestFlush(HRegion r, FlushLifeCycleTracker tracker) {
   public boolean requestFlush(HRegion r, List<byte[]> families,
       FlushLifeCycleTracker tracker) {
     synchronized (regionsInQueue) {
-      if (!regionsInQueue.containsKey(r)) {
-        // This entry has no delay so it will be added at the top of the flush
-        // queue. It'll come out near immediately.
-        FlushRegionEntry fqe = new FlushRegionEntry(r, families, tracker);
-        this.regionsInQueue.put(r, fqe);
-        this.flushQueue.add(fqe);
-        r.incrementFlushesQueuedCount();
-        return true;
-      } else {
-        tracker.notExecuted("Flush already requested on " + r);
-        return false;
+      if (regionsInQueue.containsKey(r)) {
+        FlushRegionEntry existFqe = regionsInQueue.get(r);
+        // if a delayed one exists, just remove it
+        if (existFqe.whenToExpire > existFqe.createTime) {
```

Comment:
Can we add a method in FlushRegionEntry  which check and return a boolean for this?

Reply:
It is possible that when we are here,  existFqe.whenToExpire  < System.currentTime ?  In such a case, it would be better to keep that FRE rather than adding a new one which is gonna be placed after this removed entry in Q?  Can check once pls?

Reply:
Good point, will fix, thanks.